examples/tree_2d_dgsem/elixir_euler_riemannproblem_quadrants_amr.jl

```diff
@@ -0,0 +1,159 @@
+    return prim2cons(prim, equations)
+end
+initial_condition = initial_condition_rp
+
+# See Section 2.3 of the reference below for a discussion of robust
+# subsonic inflow/outflow boundary conditions.
+#
+# - Jan-Reneé Carlson (2011)
+#   Inflow/Outflow Boundary Conditions with Application to FUN3D.
+#   [NASA TM 20110022658](https://ntrs.nasa.gov/citations/20110022658)
+@inline function boundary_condition_subsonic(u_inner, orientation::Integer,
+                                             normal_direction, x, t,
+                                             surface_flux_function,
+                                             equations::CompressibleEulerEquations2D)
+    rho_loc, v1_loc, v2_loc, p_loc = cons2prim(u_inner, equations)
+
+    p_loc = pressure(initial_condition_rp(x, t, equations), equations)
+
+    prim = SVector(rho_loc, v1_loc, v2_loc, p_loc)
+    u_surface = prim2cons(prim, equations)
+
+    return Trixi.flux(u_surface, orientation, equations)
+end
+
+boundary_conditions = (x_neg = boundary_condition_subsonic,
+                       x_pos = boundary_condition_do_nothing,
+                       y_neg = boundary_condition_subsonic,
+                       y_pos = boundary_condition_do_nothing)
+
+coordinates_min = (0.0, 0.0)
+coordinates_max = (1.0, 1.0)
+
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 4,
+                n_cells_max = 100_000,
+                periodicity = false)
+
+surface_flux = flux_hllc
+volume_flux = flux_ranocha
+
+polydeg = 3
+basis = LobattoLegendreBasis(polydeg)
+shock_indicator = IndicatorHennemannGassner(equations, basis,
+                                            alpha_max = 0.5,
+                                            alpha_min = 0.001,
+                                            alpha_smooth = true,
+                                            variable = Trixi.density)
+
+volume_integral = VolumeIntegralShockCapturingHG(shock_indicator;
+                                                 volume_flux_dg = volume_flux,
+                                                 volume_flux_fv = surface_flux)
+
+solver = DGSEM(polydeg = polydeg, surface_flux = surface_flux,
+               volume_integral = volume_integral)
+
+# Specialized function for computing coefficients of `func`,
+# here the discontinuous `initial_condition_rp`.
+#
+# Shift the outer (i.e., ±1 on the reference element) nodes passed to the 
+# `func` inwards by the smallest amount possible, i.e., [-1 + ϵ, +1 - ϵ].
+# This avoids steep gradients in elements if a discontinuity is right at a cell boundary,
+# i.e., if the jump location `x_jump` is at the position of an interface which is shared by 
+# the nodes x_{e-1}^{(i)} = x_{e}^{(1)}.
+#
+# In particular, this results in the typically desired behaviour for 
+# initial conditions of the form
+#           { u_1, if x <= x_jump
+# u(x, t) = {
+#           { u_2, if x > x_jump
+function Trixi.compute_coefficients!(u, func, t,
```

Comment:
How about something like this for safety?
```suggestion
function Trixi.compute_coefficients!(u, func::typeof(initial_condition_rp), t,
```